tools/golang/docker-compose.yml

```diff
@@ -32,4 +32,4 @@ services:
       dockerfile: tools/golang/Dockerfile
       target: unit
     volumes:
-      - ../..:/go/src/github.com/apache/trafficcontrol
```

Comment:
wait why the removal of this volume?

Reply:
Couple reasons

1. Because the Dockerfile is already `ADD`ing the whole context to that path.
2. Using that volume meant that some of the temporary files being created were being output into the source directory owned by `root:root` with permissions `-rw-------` and in general if we did things like `go fmt` it'd modify the actual source when ran. Seemed dangerous.

Reply:
well it makes me rebuild the container everytime I want to run it locally which is super annoying as it takes forever to build since we do `go get`

Reply:
Well it shouldn't be `go get`-ing *everything* anymore - that was actually causing problems because it was fetching vendored dependencies (and some of ours are so old they don't exist anymore). So now it should only be fetching things from `golang.org/x/`.

Reply:
still takes forever and will probably make me not use it locally :/

Reply:
Why would you use Docker to use it locally?

I clocked the `go get` step at 20 seconds running locally, which I wouldn't consider 'forever' but also note that it would still need to do that step if the volume still existed. The only thing that could be faster is that `ADD` line for the repository that doesn't need to exist if it's in a volume, and I didn't put that in there. Not sure why it was in there, tbh.

So deleting this line didn't make anything slower - although that doesn't mean it couldn't help it be faster.

Reply:
I like/prefer using docker locally for my work as it matches what will be seen in ci and will be consistent but thats personally preference.

We dont change our dependencies much so my thought of putting it together was I would have to just build the container once in awhile as dependencies change and be able to quickly verify all tests in the repo are passing with passing in the current code as a volume. 

my review isnt really voting anywho 🤷‍♂ so you can leave it as is

Reply:
Ohh, I see. Well lemme ask you this: why do we need the `ADD` at all if there's going to be a volume? Do we? Can I just get rid of it?

Reply:
Ahh I get what your saying now :) yeah we should be able to remove the ADD since its a volume.

Reply:
No actually we can't unless we move the `go get`s to runtime. Because the volume isn't available during the build.